Bug Import: yaml type not fully supported

When I'm using a regular form-post with a yaml file express reports the file type to be `application/x-yaml` while the [state.js#L25](https://github.com/Unleash/unleash/blob/master/lib/routes/admin-api/state.js#L25) is checking for `text/yaml`.

I also notice that the unleash crashes when this happens:

```
[2019-03-15T16:07:13.823] [INFO] HTTP - 200 POST /api/admin/state/import
application/x-yaml
[ERROR] SyntaxError: Unexpected token e in JSON at position 1
    at JSON.parse (<anonymous>)
    at StateController.import (/home/ivaosthu/code/unleash/unleash/lib/routes/admin-api/state.js:30:29)
    at Layer.handle [as handle_request] (/home/ivaosthu/code/unleash/unleash/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/ivaosthu/code/unleash/unleash/node_modules/express/lib/router/route.js:137:13)
    at Array.<anonymous> (/home/ivaosthu/code/unleash/unleash/node_modules/multer/lib/make-middleware.js:53:37)
    at listener (/home/ivaosthu/code/unleash/unleash/node_modules/on-finished/index.js:169:15)
    at onFinish (/home/ivaosthu/code/unleash/unleash/node_modules/on-finished/index.js:100:5)
    at callback (/home/ivaosthu/code/unleash/unleash/node_modules/ee-first/index.js:55:10)
    at IncomingMessage.onevent (/home/ivaosthu/code/unleash/unleash/node_modules/ee-first/index.js:93:5)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9)
Program node server.js exited with code 1

Starting child process with 'node server.js'
?
```

/cc @Kouzukii

--- a/lib/routes/admin-api/state.js
+++ b/lib/routes/admin-api/state.js
@@ -4,6 +4,7 @@ const Controller = require('../controller');
 const { ADMIN } = require('../../permissions');
 const extractUser = require('../../extract-user');
 const { handleErrors } = require('./util');
+const mime = require('mime');
 const YAML = require('js-yaml');
 const moment = require('moment');
 const multer = require('multer');
@@ -20,18 +21,18 @@ class StateController extends Controller {
         const userName = extractUser(req);
         const { drop } = req.query;
 
-        let data;
-        if (req.file) {
-            if (req.file.mimetype === 'text/yaml') {
-                data = YAML.safeLoad(req.file.buffer);
-            } else {
-                data = JSON.parse(req.file.buffer);
-            }
-        } else {
-            data = req.body;
-        }
-
         try {
+            let data;
+            if (req.file) {
+                if (mime.lookup(req.file.originalname) === 'text/yaml') {
+                    data = YAML.safeLoad(req.file.buffer);
+                } else {
+                    data = JSON.parse(req.file.buffer);
+                }
+            } else {
+                data = req.body;
+            }
+
             await this.config.stateService.import({
                 data,
                 userName,

